src/zcash/ProvingServiceClient.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2017 The Zcash developers
```

Comment:
How come in this file we don't have to check if ZMQ is enabled?

Reply:
Because includes of this file are wrapped with `#if ENABLE_PROVING_SERVICE`, which per `configure.ac` is only set if ZMQ is enabled (if someone tries to enable proving service support with ZMQ disabled, the former is overridden).

Reply:
If we added WebSocket support to `ProvingServiceClient` (or any other transport), then yes we would need to wrap the ZMQ parts as appropriate. But that can be done then (if ever).